### PR TITLE
PR: Add Pylint message name to code analysis pane and refine format

### LINKED ISSUE #12735: Change output of code analysis to show the full names of the pylint messages
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
 We should change the output of the code analysis to show full names of the messages as they are clearer to read when used for message control in the code, and using the numeric versions for that purpose is deprecated. E.g. pointless-statement for W0104.

### What steps reproduce the problem?

1. 
2. 
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* ~~[x] Added unit test(s) covering the changes (if testable)~~ Not sure if needed if generally covered by existing tests; checking for a specific format in the tests might make them too fragile to Spyder/pylint chnages

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Per discussion with @juanis2112 and @ccordoba12 , we've added the canonical name to each Pylint message, which more concisely and human-readibly identifies it and is the preferred form for `# pylint: disable` directives (error codes being deprecated for that purpose, with a Pylint warning displayed). As part of that, I did some minor cleanup on the text format in general, and a bit of refactoring on the relevant code.

To note, the fact that the Pylint message data was inexplicably stored as a tuple rather than a dict made adding this and maintaining backward compatibility much more work than it needed to be, while also having the caveat that if a user upgrades to a newer version of Spyder, successfully opened it, runs Pylint, downgrades to an older version, and then opens the pane again, an error message will be displayed trying to load the archived Pylint data. This is unavoidable due to the poor design of the existing code and data structures, but should be a very niche case and quite unlikely to occur in practice, as a user would typically not downgrade Spyder unless a newer version failed to load at all or had a serious problem, making any running of Pylint unlikely at best.

Finally, I bumped up the minimum Pylint version up to 1.0, which was released 7 years ago, as required for this PR to fully work.

Before:

image

After:

image

### Issue(s) Resolved

Fixes #12735

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: CAM-Gerlach

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
@goanpeca @ccordoba12 Given the Python 2 builds are not even being widely distributed anymore, and I have no idea why it doesn't support tuple unpacking or what to replace it with there, what needs to be done to fix the issue identified by the CIs?

**Comment by goanpeca** [comment]
Hi @CAM-Gerlach, thanks for working on this.

You could do something like:

```python
message_data = tuple(list(message_data) +  [None])
```

**Comment by ccordoba12** [comment]
This looks pretty nice @CAM-Gerlach! Thanks for your help with it!

I have one UI suggestion for now: what about if we simply remove the codes and leave things as

    (message-name) line #: Message

?

I think using the codes seems now unnecessary and even a bit unhelpful.

Pinging @juanis2112 about this too, to know what she thinks about my suggestion.

**Comment by juanis2112** [comment]
Thanks for working on this @CAM-Gerlach. I thought we said we were leaving the codes for the classification in the pane @ccordoba12

**Comment by ccordoba12** [comment]
> I thought we said we were leaving the codes for the classification in the pane @ccordoba12

Oh yeah, I forgot about that. Then I think it's fine :+1:

**Comment by CAM-Gerlach** [comment]
Yeah, what @juanis2112 said was also my understanding, which is why I left them.

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 1]
```diff
@@ -94,13 +94,17 @@ def refresh(self):
...
             title_item = QTreeWidgetItem(self, [title], QTreeWidgetItem.Type)
             title_item.setIcon(0, icon)
             if not messages:
                 title_item.setDisabled(True)
             modules = {}
-            for module, lineno, message, msg_id in messages:
+            for message_data in messages:
+                # If message data is legacy version without message_name
+                if len(message_data) == 4:
+                    message_data = (*message_data, None)
```
Comment (CAM-Gerlach) [inline spyder/plugins/pylint/widgets/pylintgui.py:106]:
```suggestion
                    message_data = tuple(list(message_data) + [None])
```

**Comment by ccordoba12** [comment]
@goanpeca, please help me to review this one.

**Comment by CAM-Gerlach** [comment]
Thanks, @goanpeca , added. I just don't have the mental bandwidth to support an EoL language I've never actually used for my own work anymore.

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 2]
```diff
@@ -94,13 +94,17 @@ def refresh(self):
...
             title_item.setIcon(0, icon)
             if not messages:
                 title_item.setDisabled(True)
             modules = {}
-            for module, lineno, message, msg_id in messages:
+            for message_data in messages:
+                # If message data is legacy version without message_name
+                if len(message_data) == 4:
+                    message_data = tuple(list(message_data) + [None])
+                module, lineno, message, msg_id, message_name = message_data
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:107]:
```suggestion

                module, lineno, message, msg_id, message_name = message_data

```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 3]
```diff
@@ -128,10 +132,16 @@ def refresh(self):
...
                 if len(msg_id) > 1:
-                    text = "[%s] %d : %s" % (msg_id, lineno, message)
-                else:
-                    text = "%d : %s" % (lineno, message)
-                msg_item = QTreeWidgetItem(parent, [text], QTreeWidgetItem.Type)
+                    if not message_name:
+                        message_string = "{msg_id} "
+                    else:
+                        message_string = "{msg_id} ({message_name}) "
+                message_string += "line {lineno}: {message}"
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:139]:
```suggestion

                message_string += "line {lineno}: {message}"

```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 4]
```diff
@@ -421,20 +426,28 @@ def finished(self, exit_code, exit_status):
...
+            items = {}
+            idx_0 = 0
+            idx_1 = 0
+            key_names = ["msg_id", "message_name", "line_nb", "message"]
+            for key_idx, key_name in enumerate(key_names):
+                if key_idx == len(key_names) - 1:
+                    idx_1 = len(line)
+                else:
+                    idx_1 = line.find(":", idx_0)
+                if idx_1 < 0:
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:438]:
```suggestion

                if idx_1 < 0:

```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 5]
```diff
@@ -421,20 +426,28 @@ def finished(self, exit_code, exit_status):
...
+                    break
+                item = line[(idx_0):idx_1]
+                if not item:
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:441]:
```suggestion

                if not item:

```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 6]
```diff
@@ -421,20 +426,28 @@ def finished(self, exit_code, exit_status):
...
+                    break
+                if key_name == "line_nb":
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:443]:
```suggestion

                if key_name == "line_nb":

```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 7]
```diff
@@ -421,20 +426,28 @@ def finished(self, exit_code, exit_status):
...
+                    item = int(item.split(',')[0])
+                items[key_name] = item
```
Comment (goanpeca) [inline spyder/plugins/pylint/widgets/pylintgui.py:445]:
```suggestion

                items[key_name] = item

```

**Comment by CAM-Gerlach** [comment]
Thanks @goanpeca ! I applied all the suggestions, and since they were minor style changes, I just fixuped them into to the original commit while leaving the Py2 compat one as seperate so its easy to revert or drop when merging to Py 3.x `master`.

**Comment by ccordoba12** [comment]
@goanpeca, is this ready?

**Comment by goanpeca** [comment]
Yes this one is ready!

**Pull request merged**